### PR TITLE
[#Refactor Code] refactor code in active_support callbacks

### PR DESCRIPTION
**Pull request by narutosanjiv**

### CODE REVIEW

File: activesupport/lib/active_support/callbacks.rb [hunk 1]
```diff
@@ -341,14 +328,18 @@ def self.conditional(callback_sequence, user_callback, user_conditions)
 
         def self.simple(callback_sequence, user_callback)
           callback_sequence.around do |env, &run|
-            user_callback.call(env.target, env.value) {
+            run_user_callback(user_callback, env, run)
+          end
+        end
+        private_class_method :simple
+
+        def self.run_user_callback(user_callback, env, run)
```
Comment (egilburg) [inline activesupport/lib/active_support/callbacks.rb]: Should be marked as private too?

**Comment by rafaelfranca** [comment]
It seems good but this code is performance critical. Could you benchmark you code before and after this change?

**Comment by narutosanjiv** [comment]
@rafaelfranca  Ok will do it

**Comment by repinel** [comment]
The refactored code seems to be a little bit slower using this benchmark

https://gist.github.com/repinel/90746f0f5ae69cc3903a

Before:
```
Calculating -------------------------------------
 halting conditional     6.119k i/100ms
             halting     7.313k i/100ms
-------------------------------------------------
 halting conditional     87.515k (±17.1%) i/s -    428.330k
             halting    103.411k (±19.9%) i/s -    504.597k
```

After:
```
Calculating -------------------------------------
 halting conditional     6.760k i/100ms
             halting     7.758k i/100ms
-------------------------------------------------
 halting conditional     82.909k (±15.1%) i/s -    405.600k
             halting    101.854k (±17.9%) i/s -    504.270k
```

File: activesupport/lib/active_support/callbacks.rb [hunk 2]
```diff
@@ -311,11 +306,7 @@ def self.halting(callback_sequence, user_callback)
             if env.halted
               run.call env
             else
-              user_callback.call(target, value) {
-                env = run.call env
-                env.value
-              }
-              env
+              run_user_callback(user_callback, env, run)
```
Comment (repinel) [inline activesupport/lib/active_support/callbacks.rb]: With this change, you can also remove the variables `target` and `value`.

**Comment by narutosanjiv** [comment]
@repinel Which tool you have used for benchmarking?.

**Comment by rafaelfranca** [comment]
@narutosanjiv See https://gist.github.com/repinel/90746f0f5ae69cc3903a

**Pull request closed without merging**